common/src/main/java/com/wynntils/screens/settings/widgets/FeatureSettingWidget.java

```diff
@@ -0,0 +1,306 @@
+        final float padding = settingsScreen.width / 100f;
+
+        float renderY = settingsScreen.getBarHeight() + padding;
+
+        final float xOffset = this.width / 35f;
+
+        hoveredConfigElement = null;
+
+        Overlay lastOverlayConfigTitle = null;
+
+        for (int i = scrollIndexOffset; i < configWidgets.size(); i += 2) {
+            ConfigOptionElement configWidgetLeft = configWidgets.get(i);
+            ConfigOptionElement configWidgetRight = configWidgets.size() > i + 1 ? configWidgets.get(i + 1) : null;
+            float renderHeight = this.height / 4f;
+
+            float fullWidth = this.width - xOffset * 1.5f;
+            float renderWidth = fullWidth / 2 - padding;
+
+            if (configWidgetLeft.getConfigHolder().getParent() instanceof Overlay overlay
+                    && configWidgetLeft.getConfigHolder().getParent() != lastOverlayConfigTitle) {
+                Font font = FontRenderer.getInstance().getFont();
+                float textRenderHeight = font.lineHeight + padding;
+
+                if (renderY + textRenderHeight > this.y + this.height) break;
+
+                lastOverlayConfigTitle = overlay;
+
+                FontRenderer.getInstance()
+                        .renderAlignedTextInBox(
+                                poseStack,
+                                overlay.getTranslatedName(),
+                                0,
+                                this.width,
+                                renderY,
+                                0,
+                                CommonColors.WHITE,
+                                FontRenderer.TextAlignment.CENTER_ALIGNED,
+                                FontRenderer.TextShadow.OUTLINE);
+                renderY += textRenderHeight;
+            }
+
+            if (renderY + renderHeight > this.y + this.height) break;
+
+            configWidgetLeft.render(
+                    poseStack, xOffset, renderY, renderWidth, renderHeight, mouseX, mouseY, partialTick);
+
+            float actualRenderX = this.x + xOffset;
+            float actualRenderY = this.y + renderY;
+
+            if (isMouseOverConfigWidget(actualRenderX, actualRenderY, renderWidth, renderHeight, mouseX, mouseY)) {
+                this.hoveredConfigElement = configWidgetLeft;
+            }
+
+            if (configWidgetRight != null) {
+                actualRenderX = this.x + xOffset + fullWidth / 2;
+                renderWidth = fullWidth / 2;
+                float renderX = xOffset + fullWidth / 2;
+
+                if (isMouseOverConfigWidget(actualRenderX, actualRenderY, renderWidth, renderHeight, mouseX, mouseY)) {
+                    this.hoveredConfigElement = configWidgetRight;
+                }
+
+                if (configWidgetRight.getConfigHolder().getParent() instanceof Overlay
+                        && configWidgetRight.getConfigHolder().getParent() != lastOverlayConfigTitle) {
+                    renderY += renderHeight + padding;
```

Comment:
I'm not sure I understand the if condition here. At what circumstance do we want/need to render the right widget to the left?

Reply:
When there is an overlay, we need to render the title. If the left config field is the first for the overlay, it is easy case. If it is the right, we shift the right config title to be the left one, also rendering the title next iteration.